test/core/pthread/test_pthread_keepalive.c

```diff
@@ -0,0 +1,63 @@
+#include <assert.h>
+#include <stdatomic.h>
+#include <stdbool.h>
+#include <pthread.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include <emscripten/emscripten.h>
+#include <emscripten/eventloop.h>
+#include <emscripten/console.h>
+#include <emscripten/proxying.h>
+
+pthread_t t;
+pthread_cond_t cond = PTHREAD_COND_INITIALIZER;
+pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+
+void timeout(void* arg) {
+  _emscripten_errf("timeout");
+  pthread_mutex_lock(&mutex);
+  pthread_cond_signal(&cond);
+  pthread_mutex_unlock(&mutex);
+}
+
+void* thread_main(void* arg) {
+  // Keep the thread runtime alive for now.
+  _emscripten_errf("thread_main");
+  emscripten_runtime_keepalive_push();
+  emscripten_set_timeout(timeout, 0, NULL);
+  return NULL;
+}
+
+void say_hello(void* arg) {
+  _emscripten_errf("say_hello");
+}
+
+void keeplive_pop(void* arg) {
+  _emscripten_errf("keeplive_pop");
+  // After this the called, thread should exit (become joinable).
+  emscripten_runtime_keepalive_pop();
+}
+
+int main() {
+  printf("main\n");
+  pthread_mutex_lock(&mutex);
+  int rc = pthread_create(&t, NULL, thread_main, NULL);
+  assert(rc == 0);
+
+  // Wait until the thread is running from the event loop
+  pthread_cond_wait(&cond, &mutex);
+
+  // Run something from the thread's event loop
+  emscripten_proxy_sync(emscripten_proxy_get_system_queue(), t, &say_hello, NULL);
+
+  // Run keeplive_pop, causing the thread to exit gracefully
+  emscripten_proxy_sync(emscripten_proxy_get_system_queue(), t, &keeplive_pop, NULL);
```

Comment:
How does this make the thread joinable given that `executeNotifiedProxyingQueue` does not use `callUserCallback`? Is it because we happen to be using the system event queue and that has some extra handling somewhere?

Reply:
This change includes that change to have executeNotifiedProxyingQueue use callUserCallback.